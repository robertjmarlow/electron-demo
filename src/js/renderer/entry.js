const { ipcRenderer } = require('electron');
const $ = require('jquery');
const os = require('os');
require('../../css/main.css');

$(document).ready(function() {
  ipcRenderer.on('getFileContents', function(event, arg) {
    $('#file-contents').append('<div class="fileContents">' + arg.toString().replace(/\n/g, '<br/>').replace(/ /g, '&nbsp') + '</div>');
  });

  ipcRenderer.on('getDirectoryContents', function(event, arg) {
    // yes, this was broken in the last checkin...
    $('#dir-contents').append('<div class="fileList">' + arg + '</div>');
  });

  ipcRenderer.send('getFileContents', './main.js');
  ipcRenderer.send('getDirectoryContents', '.');

  $('#os-stuff')
    .append('<div>Your OS is: <span class="big">' + os.platform() + ' ' + os.release() + '</span></div>')
    .append('<div>Your uptime (in seconds) is: <span class="big">' + os.uptime() + '</span></div>')
});
