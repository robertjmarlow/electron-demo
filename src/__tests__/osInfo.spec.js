import React from 'react';
import renderer from 'react-test-renderer';
import OsInfo from '../jsx/osInfo';

jest.mock('os');

test('Displays OS info', () => {
  const tree = renderer.create(
    <OsInfo/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
