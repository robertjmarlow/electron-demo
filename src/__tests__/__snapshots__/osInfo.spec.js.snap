exports[`test Displays OS info 1`] = `
<div>
  <span
    className="big">
    Platform:
  </span>
   
  Platform
   
  Arch
   
  <span
    className="big">
    Uptime:
  </span>
   
  9001
</div>
`;
